src/core/Akka.Persistence/Snapshot/SnapshotStore.cs

```diff
@@ -43,43 +43,38 @@ protected SnapshotStore()
                 config.GetTimeSpan("circuit-breaker.reset-timeout"));
         }
 
-        /// <summary>
-        /// TBD
-        /// </summary>
-        /// <param name="message">TBD</param>
-        /// <returns>TBD</returns>
+        /// <inheritdoc/>
         protected sealed override bool Receive(object message)
         {
             return ReceiveSnapshotStore(message) || ReceivePluginInternal(message);
         }
 
         private bool ReceiveSnapshotStore(object message)
         {
-            if (message is LoadSnapshot)
-            {
-                var msg = (LoadSnapshot) message;
+            var senderPersistentActor = Sender; // Sender is PersistentActor
+            var self = Self; //Self MUST BE CLOSED OVER here, or the code below will be subject to race conditions
```

Comment:
These variables should be closed over here, in order to prevent racing conditions